pageserver/src/tenant/timeline.rs

```diff
@@ -614,6 +614,31 @@ impl Timeline {
     pub async fn compact(self: &Arc<Self>, ctx: &RequestContext) -> anyhow::Result<()> {
         const ROUNDS: usize = 2;
 
+        static CONCURRENT_COMPACTIONS: once_cell::sync::Lazy<tokio::sync::Semaphore> =
```

Comment:
I'm not sure if this is a good place to limit compaction concurrency b/c besides compaction triggered from intervals, there are also the ones triggered manually and by checkpoint. We probably want to give higher priority for this kinds of compaction tasks, and it would be better to move the concurrency limitation to the "schedule compaction by interval" part of the code?

Reply:
Manual compactions only happen within tests as far as I know. Similarly checkpointing manually only happens in tests -- in production `WalReceiver` calls `Timeline::check_checkpoint_distance` which can flush, and that will create work for the next periodic compaction.